plugins/scaffolder-backend-module-gitea/src/actions/gitea.ts

```diff
@@ -0,0 +1,236 @@
+import { Config } from '@backstage/config';
+import {
+  getGiteaRequestOptions,
+  GiteaIntegrationConfig,
+  ScmIntegrationRegistry,
+} from '@backstage/integration';
+import {
+  createTemplateAction,
+  getRepoSourceDirectory,
+  initRepoAndPush,
+  parseRepoUrl,
+} from '@backstage/plugin-scaffolder-node';
+import { examples } from './gitea.examples';
+import fetch, { RequestInit, Response } from 'node-fetch';
+import crypto from 'crypto';
+
+const createGiteaProject = async (
+  config: GiteaIntegrationConfig,
+  options: {
+    projectName: string;
+    // parent: string;
+    owner?: string;
+    description: string;
+  },
+): Promise<void> => {
+  const { projectName, owner, description } = options;
+
+  const fetchOptions: RequestInit = {
+    method: 'POST',
+    body: JSON.stringify({
+      name: projectName,
+      description,
+      owner,
+    }),
+    headers: {
+      ...getGiteaRequestOptions(config).headers,
+      'Content-Type': 'application/json',
+    },
+  };
+
+  // TODO
+  // Create a repository in Gitea
+  // API request: https://gitea.com/api/swagger#/user/createCurrentUserRepo
+  const response: Response = await fetch(
+    `${config.baseUrl}/api/v1/user/repos`,
+    fetchOptions,
+  );
+  if (response.status !== 201) {
+    throw new Error(
+      `Unable to create repository, ${response.status} ${
+        response.statusText
+      }, ${await response.text()}`,
+    );
+  }
+};
+
+const generateCommitMessage = (
+  config: Config,
+  commitSubject?: string,
+): string => {
+  const changeId = crypto.randomBytes(20).toString('hex');
+  const msg = `${
+    config.getOptionalString('scaffolder.defaultCommitMessage') || commitSubject
+  }\n\nChange-Id: I${changeId}`;
+  return msg;
+};
+
+/**
+ * Creates a new action that initializes a git repository using the content of the workspace.
+ * and publishes it to a Gitea instance.
+ * @public
+ */
+export function createPublishGiteaAction(options: {
+  integrations: ScmIntegrationRegistry;
+  config: Config;
+}) {
+  const { integrations, config } = options;
+
+  return createTemplateAction<{
+    repoUrl: string;
+    description: string;
+    defaultBranch?: string;
+    gitCommitMessage?: string;
+    gitAuthorName?: string;
+    gitAuthorEmail?: string;
+    sourcePath?: string;
+  }>({
+    id: 'publish:gitea',
+    description:
+      'Initializes a git repository using the content of the workspace, and publishes it to Gitea.',
+    examples,
+    schema: {
+      input: {
+        type: 'object',
+        required: ['repoUrl'],
+        properties: {
+          repoUrl: {
+            title: 'Repository Location',
+            type: 'string',
+          },
+          description: {
+            title: 'Repository Description',
+            type: 'string',
+          },
+          defaultBranch: {
+            title: 'Default Branch',
+            type: 'string',
+            description: `Sets the default branch on the repository. The default value is 'master'`,
```

Comment:
seems to actually be "main", not "master"

Reply:
Fixed with 9e85efc22e